### ISSUE TITLE
Issue with Current Blocked Items Chart

### ISSUE THREAD

**Issue by ChristopherMank** [nbrown02/FlowViz#147]
**Describe the bug**
The Current Blocked Items chart states in the wiki that it shows the items that are currently identified as blocked and how long (in calendar days) since they were last updated. That is what I see is showing in the chart. However, the wiki also states that in the example, work item 8 is currently ‘blocked’ – The Current Blocked Items chart would highlight how long it has had the blocked tag for. The PBI report also states that it shows the currently blocked items and how long they been blocked for. 

These are contradicting. From what I see, the report is actually showing how long since they were last modified and not for how long they have been blocked for. I feel the latter is more valuable, showing how long they have been blocked for. Can the wiki be cleaned up and the chart updated to reflect that actual behavior?

**Comment by nbrown02** [comment]
Thanks for sharing this. Yeah it initially started as how long it has been blocked for...then I realised this was harder to do and went with time since last modified (as this was easier). I'll take a look again next week to see if I can do how long blocked for and reply via this once I find a solution...

**Comment by nbrown02** [comment]
@ChristopherMank can you give [this version](https://github.com/nbrown02/FlowViz/raw/refs/heads/main/Screenshots/FlowViz%20-%20new%20version.pbit) a try?

**Comment by ChristopherMank** [comment]
@nbrown02 Thanks for sending a new version. Here is what the chart looks like with the changes. I'm not sure where it's getting 337d for each of them. I reviewed each work item and have a table below of the total # of days they have been blocked and how long it's been actively blocked (I wasn't sure which one you are attempting to report on). 

![image](https://github.com/user-attachments/assets/2b9119be-6f98-44d6-b25e-38d0dc582368)


| ID     | TOTAL | CURRENT |
|--------|-------|---------|
| 203420 | 51d   | 24d     |
| 207144 | 91d   | 82d     |
| 209703 | 67d   | 61d     |
| 214591 | 40d   | 40d     |
| 214592 | 12d   | 6d      |

**Comment by nbrown02** [comment]
Could you send me a screenshot of WorkItems Blocked3 table? I think i know what the issue is but don’t have any test data so flying blind!Thanks,Nick Sent from my iPhoneOn 30 Dec 2024, at 14:21, Christopher Mank ***@***.***> wrote:﻿
@nbrown02 Thanks for sending a new version. Here is what the chart looks like with the changes. I'm not sure where it's getting 337d for each of them. I reviewed each work item and have a table below of the total # of days they have been blocked and how long it's been actively blocked (I wasn't sure which one you are attempting to report on).
image.png (view on web)



ID
TOTAL
CURRENT




203420
51d
24d


207144
91d
82d


209703
67d
61d


214591
40d
40d


214592
12d
6d

**Comment by ChristopherMank** [comment]
Here is a screenshot of the table. I sorted the second col descending as there are a lot of rows in here.

![image](https://github.com/user-attachments/assets/e011b831-83a9-4f07-8fe7-eca30ba50271)

I also filtered by one of the items I had previously listed. I can provide more screenshots/data if that would help. Thanks!

![image](https://github.com/user-attachments/assets/864b24af-ccc1-4ae0-bd68-fcca468a72c1)

**Comment by nbrown02** [comment]
@ChristopherMank Thanks for sharing. 

The issue last time was me taking the max for all work item IDs, rather than per work item ID, can you give [this version](https://github.com/nbrown02/FlowViz/raw/refs/heads/main/Screenshots/FlowViz%20-%20new%20version.pbit) a try?

One point to note, if an item has been blocked for longer than the 1st Jan 2024 it will only ever show up to that date due to the data that is loaded being limited...

**Comment by ChristopherMank** [comment]
Thanks! Here is the updated chart. The numbers have changed but still doesn't look correct. The items listed have been blocked for longer than the time indicated.

![image](https://github.com/user-attachments/assets/87e88a5c-e16d-4f44-9341-c6057ccd2afb)

For example, ID 207144 is listed as being blocked for 13 days. However, here is what it shows in ADO. It was blocked on 9/10 and unblocked on 9/19 (9d). Then it was blocked again on 10/9 and has been blocked since (82d as of today 12/30). I was expecting to see either 82d (current days blocked since last blocked) or 91d (total days blocked total). 

![image](https://github.com/user-attachments/assets/e925fc00-7668-409a-a7dd-c80ba51025f2)

If it helps, here is what the WorkItems Blocked3 table looks like:

![image](https://github.com/user-attachments/assets/c55910ae-fff1-456e-b680-6b3a7f591d41)

**Comment by nbrown02** [comment]
In the WorkItems FlowEfficiency table, is there a column that contains that date/time it was blocked?

**Comment by nbrown02** [comment]
@ChristopherMank - I think I've got it (or certainly closer!), can you give [this version](https://github.com/nbrown02/FlowViz/raw/refs/heads/main/Screenshots/FlowViz%20-%20new%20version.pbit) a try?

**Comment by ChristopherMank** [comment]
That fixed it! This looks exactly as I was expecting. Thank you so much for your prompt replies!

![image](https://github.com/user-attachments/assets/c3ed56a2-8b57-48b3-8f0f-e877dbe5b685)

![image](https://github.com/user-attachments/assets/8feff963-3eff-41a7-abac-08cbf38f33f3)

**Comment by nbrown02** [comment]
> That fixed it! This looks exactly as I was expecting. Thank you so much for your prompt replies!
> 
> 
> 
> ![image](https://github.com/user-attachments/assets/c3ed56a2-8b57-48b3-8f0f-e877dbe5b685)
> 
> 
> 
> ![image](https://github.com/user-attachments/assets/8feff963-3eff-41a7-abac-08cbf38f33f3)
> 
> 

Fantastic! Just need to merge into main now :)
